lib/OpenQA/WebAPI/Controller/API/V1/Comment.pm

```diff
@@ -138,8 +139,11 @@ sub create ($self) {
             user_id => $self->current_user->id
         });
 
-    eval { $comment->handle_special_contents($self) };
-    return $self->render(json => {error => $@}, status => 400) if $@;
+    try { $comment->handle_special_contents($self) }
+    catch ($e) {
+        undef $txn_guard;
+        return $self->render(json => {error => $e}, status => 400);
+    }
```

Comment:
```suggestion
    eval { $comment->handle_special_contents($self) };  # using eval as the txn guard interacts badly with try-catch
    return $self->render(json => {error => $@}, status => 400) if $@;
```

Reply:
> The advantage would be that we can filter for `$@` usage, only leaving `eval` where we are not using the error message.

Let me demonstrate that in
* #6303

how do you like that?

Reply:
The case where we need to undefine the txn guard is still worse.

I also don't see it as necessity to enforce use of try-catch and thus filtering doesn't seem very useful. As long as we don't use `Try::Tiny` anymore which has counter-intuitive behavior for everyone coming from different programming languages we're good.